e2e-tests/demand-backup-eks-credentials-irsa/run

```diff
@@ -0,0 +1,135 @@
+#!/bin/bash
+
+set -o errexit
+
+test_dir=$(realpath $(dirname $0))
+. ${test_dir}/../functions
+set_debug
+# EKS cluster should be run without the policy AmazonS3FullAccess
+# This policy makes the test false passed.
+
+if [ $EKS -ne 1 ]; then
+	echo "Skip the test. We run it for EKS only "
+	exit 0
+fi
+cluster="some-name-rs0"
+
+desc "get cluster oidc"
+eks_cluster=$(kubectl config view --minify -o jsonpath='{.contexts[0].context.cluster}' | awk -F/ '{print $NF}')
+IFS='.' read -r eks_cluster_name eks_cluster_region _ <<<"$eks_cluster"
+
+eks_cluster_oidc=$(aws eks describe-cluster --name $eks_cluster_name --region=$eks_cluster_region --query "cluster.identity.oidc.issuer" --output text | sed 's|https://||')
+policy_arn="arn:aws:iam::119175775298:policy/operator-testing-access-s3"
+role_name="$cluster-psmdb-access-s3-bucket"
+
+# Create policy. Already done, we don't need to do it every time. But all steps should be illustrated in the test
+#aws iam create-policy --policy-name operator-testing-allow-access-s3 --policy-document file://conf/s3-bucket-policy.json
+
+desc "create role"
+jq --arg eks_cluster_oidc "$eks_cluster_oidc" \
+	'.Statement[0].Principal.Federated = "arn:aws:iam::119175775298:oidc-provider/\($eks_cluster_oidc)" |
+    .Statement[0].Condition.StringEquals[($eks_cluster_oidc + ":aud")] = "sts.amazonaws.com"' \
+	$test_dir/conf/template.json >$test_dir/conf/role-trust-policy.json
+
+role_arn=$(aws iam create-role \
+	--role-name "$role_name" \
+	--assume-role-policy-document file://$test_dir/conf/role-trust-policy.json \
+	--description "Allow access to s3 bucket" \
+	--query "Role.Arn" \
+	--output text)
+
+desc "connect role and policy"
+aws iam attach-role-policy --role-name "$role_name" --policy-arn $policy_arn
+
+create_infra "$namespace"
+
+desc "create secrets and start client"
+kubectl_bin apply \
+	-f "$conf_dir/secrets.yml" \
+	-f "$conf_dir/client.yml"
+
+desc "create first PSMDB cluster $cluster"
+apply_cluster $test_dir/conf/$cluster.yml
+
+desc 'check if all 3 Pods started'
+wait_for_running $cluster 3
+
+desc 'check if service and statefulset created with expected config'
+compare_kubectl statefulset/$cluster
+
+desc "update service accounts for operator and default (our cluster uses this one)"
+
+kubectl_bin annotate serviceaccount default \
+	eks.amazonaws.com/role-arn="$role_arn" \
+	--overwrite
+
+kubectl_bin annotate serviceaccount percona-server-mongodb-operator \
+	eks.amazonaws.com/role-arn="$role_arn" \
+	--overwrite
+
+desc "restart operator and cluster"
```

Comment:
Hi,
To consider: cover/check use-case when at first serviceaccount is created and then used during start of the first cluster.

Reply:
it works in this case too.